### PR TITLE
modify dfm service for dependency.

### PR DESCRIPTION
**Pull request by vwxyzh**
@chenkennt @superyyrrzz @hellosnow @ansyral @qinezh

### CODE REVIEW

File: src/Microsoft.DocAsCode.Dfm/DfmContextExtensions.cs [hunk 1]
```diff
@@ -48,5 +50,20 @@ public static IMarkdownContext SetBaseFolder(this IMarkdownContext context, stri
             }
             return context.CreateContext(context.Variables.SetItem(BaseFolderKey, baseFolder));
         }
+
+        public static IMarkdownContext SetDependency(this IMarkdownContext context, HashSet<string> dependency)
```
Comment (ansyral) [inline src/Microsoft.DocAsCode.Dfm/DfmContextExtensions.cs]: check null argument for `dependency`?

**Comment by superyyrrzz** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Dfm/DfmContextExtensions.cs (hunk 1 above)
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Dfm/DfmContextExtensions.cs]: check null in report dependency.

**Comment by qinezh** [comment]
:shipit:

**Comment by vwxyzh** [comment]
#449

**Comment by hellosnow** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Dfm/DfmContextExtensions.cs [hunk 2]
```diff
@@ -48,5 +50,23 @@ public static IMarkdownContext SetBaseFolder(this IMarkdownContext context, stri
...
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+            return context.CreateContext(context.Variables.SetItem(DependencyKey, dependency));
+        }
+
+        public static void ReportDependency(this IMarkdownContext context, string file)
+        {
+            var dependency = (HashSet<string>)context.Variables[DependencyKey];
+            if (dependency != null)
```
Comment (ansyral) [inline src/Microsoft.DocAsCode.Dfm/DfmContextExtensions.cs]: if null, should new a hashset?

**Comment by ansyral** [comment]
:+1:

**Pull request merged**